audformat/core/table.py

```diff
@@ -446,52 +449,73 @@ def _get_by_index(
         # Returns `df, df_is_copy`
         raise NotImplementedError()
 
-    def _index_levels_and_converters(self) -> typing.Tuple[
-            typing.Sequence[str],
-            typing.Dict[str, typing.Callable],
-    ]:  # pragma: no cover
-        # Executed when loading table from CSV file.
-        # Returns: `index_levels, index_converters`
-        raise NotImplementedError()
-
     def _load_csv(self, path: str):
 
-        usecols = []
-        dtypes = {}
         schemes = self.db.schemes
+        converters = {}
+        dtypes = {}
 
-        # index columns
+        if hasattr(self, 'type'):
+            # filewise or segmented table
+            dtypes[define.IndexField.FILE] = define.DataType.STRING
+            if self.type == define.IndexType.SEGMENTED:
+                dtypes[define.IndexField.START] = define.DataType.TIME
+                dtypes[define.IndexField.END] = define.DataType.TIME
+        else:
+            # misc table
+            dtypes = self.levels
 
-        levels, converters = self._index_levels_and_converters()
-        usecols.extend(levels)
+        # index columns
+        levels = list(dtypes)
+        dtypes = {
+            level: to_pandas_dtype(dtype)
+            for level, dtype in dtypes.items()
+        }
 
         # other columns
-
+        columns = list(self.columns)
         for column_id, column in self.columns.items():
-            usecols.append(column_id)
             if column.scheme_id is not None:
-                dtype = schemes[column.scheme_id].to_pandas_dtype()
-                # use converter if column contains dates or timestamps
-                if dtype == 'datetime64[ns]':
-                    converters[column_id] = lambda x: pd.to_datetime(x)
-                elif dtype == 'timedelta64[ns]':
-                    converters[column_id] = lambda x: pd.to_timedelta(x)
-                else:
-                    dtypes[column_id] = dtype
+                dtypes[column_id] = schemes[column.scheme_id].to_pandas_dtype()
             else:
                 dtypes[column_id] = 'str'
 
-        # read csv
+        # replace dtype with converter for dates or timestamps
+        dtypes_wo_converters = {}
+        for column_id, dtype in dtypes.items():
+            if dtype == 'datetime64[ns]':
+                converters[column_id] = lambda x: pd.to_datetime(x)
+            elif dtype == 'timedelta64[ns]':
+                converters[column_id] = lambda x: pd.to_timedelta(x)
+            else:
+                dtypes_wo_converters[column_id] = dtype
 
+        # read csv
         df = pd.read_csv(
             path,
-            usecols=usecols,
-            dtype=dtypes,
+            usecols=levels + columns,
```

Comment:
Just a random though - but do we have actually have to make sure that there is no overlap between level and column names?

Reply:
Yes, let's work on this in a separate PR.